src/main/java/com/apicatalog/jsonld/loader/LRUDocumentCache.java

```diff
@@ -0,0 +1,52 @@
+package com.apicatalog.jsonld.loader;
+
+import com.apicatalog.jsonld.JsonLdError;
+import com.apicatalog.jsonld.context.cache.LruCache;
+import com.apicatalog.jsonld.document.Document;
+
+import java.net.URI;
+import java.util.Collection;
+
+public class LRUDocumentCache implements DocumentLoader {
+
+    private final DocumentLoader documentLoader;
+
+    private final LruCache<String, Document> cache;
+
+    public LRUDocumentCache(int cacheSize) {
+        this.documentLoader = SchemeRouter.defaultInstance();
+        this.cache = new LruCache<>(cacheSize);
+    }
+
+    public LRUDocumentCache(DocumentLoader documentLoader, int cacheSize) {
+        this.documentLoader = documentLoader;
+        this.cache = new LruCache<>(cacheSize);
+    }
+
+    @Override
+    public Document loadDocument(URI url, DocumentLoaderOptions options) throws JsonLdError {
+        String key = computeCacheKey(url, options);
+        Document result = cache.get(key);
+        if (result == null) {
+            result = documentLoader.loadDocument(url, options);
+            cache.put(key, result);
+        }
+        return result;
+    }
+
+    protected String computeCacheKey(URI url, DocumentLoaderOptions options) {
```

Comment:
Just thinking, what about to implement `DocumentLoaderOptions` properly. i.e. to add hashCode() and equals() methods?

Reply:
Just to be sure, does the ordering of values in `requestProfile` matter?

Reply:
That's a very good question,  let's suppose it does matter, which makes the implementation more complicated, unfortunately. 

BTW. Looking at the [getAcceptHeader](https://github.com/filip26/titanium-json-ld/blob/a52f3920ecde13c0b381c10930570eabfdd397f8/src/main/java/com/apicatalog/jsonld/loader/DefaultHttpLoader.java#L171)  method I think there could be a bug. The profile param serialization implementation should be supported by a link to a specification, even though I would expect having multiple profiles as a very rare use-case.

Reply:
I hope I've addressed your comment correctly.

Reply:
Good job! Thanks again for contributing.